### DISCUSSION TITLE
java.lang.OutOfMemoryError: GC overhead limit exceeded

### DISCUSSION THREAD

**Discussion by stephanpauls** [cantaloupe-project/cantaloupe#491] [Q&A]
Hi,

At regular times, mostly after a month the java error ' java.lang.OutOfMemoryError: GC overhead limit exceeded'
occurs and the system crashes. I can increase the heap memory but this will only postpone the error.

Is the problem known and if so is there a better solution? 

Thanks,

Stephan Pauls

**Comment by adolski** [comment]
The error could be caused by your use case or by insufficient heap memory.

Certain use cases place more of a burden on the heap. If some request is taking a long time, like reading a non-pyramidal, non-tiled TIFF image, it may be spending that time filling up memory--a crude overgeneralization but a reminder to check that you are making efficient requests using efficient image formats.

After that is checked, the solution is generally to scale either vertically (add more heap) or horizontally (add more heap by adding more nodes). As you add more, it should happen less often, and eventually you will find the threshold at which it doesn't happen.

Are you sure the system crash is related?
